### PR TITLE
ipfs get: set correct content-type on resp

### LINKED ISSUE #1824: HTTP API -
The *get* command, e.g. ``POST /api/v0/get?arg=QmedYJNEKn656faSHaMv5UFVkgfSzwYf9u4zsYoXqgvnch``, always returns a TAR stream. However it doesn't set the Content-Type header accordingly. It should be *application/x-tar*.

### PR DESCRIPTION
**Pull request by rht**
closes #1824

### CODE REVIEW

**Comment by whyrusleeping** [comment]
mmm... i see you PRing against master... i'll allow this one though. fairly trivial.

LGTM

File: commands/http/handler.go [hunk 1]
```diff
@@ -232,7 +234,9 @@ func sendResponse(w http.ResponseWriter, r *http.Request, res cmds.Response, req
 	if _, ok := res.Output().(io.Reader); ok {
 		// set streams output type to text to avoid issues with browsers rendering
 		// html pages on priveleged api ports
-		mime = "text/plain"
+		if mime != mimeTypes[cmds.Tar] && mime != mimeTypes[cmds.Gzip] {
```
Comment (jbenet) [inline commands/http/handler.go]: maybe can make it something like:

```go
if !xssSafeMimeType(mime) {
  mime = mimeTypes[cmds.Text]
}
```

and elsewhere something like:

```go
var xssSafeMimeTypes = []string{
  mimeTypes[cmds.Tar],
  mimeTypes[cmds.GZip],
  mimeTypes[cmds.Text],
  mimeTypes[cmds.JSON],
}

func xssSafeMimeType(mime string) {
  for _, t := range xssSafeMimeTypes {
    if t == mime {
      return true
    }
  }
  return false
}
```
to make it clear _why_ this is happening, and easier to broaden the safety boundary, in a single place.

this probably could be a separate package, and may already exist somewhere.

**Comment by jbenet** [comment]
one comment above, otherwise LGTM

**Comment by whyrusleeping** [comment]
@rht update here?

File: commands/http/handler.go (hunk 1 above)
Comment (rht) [inline commands/http/handler.go]: No such package exists elsewhere yet.

**Comment by rht** [comment]
@rht doesn't have commit access

**Pull request closed without merging**